examples/dynamic/mne_var_connectivity.py

```diff
@@ -0,0 +1,215 @@
+# Each Epoch here represents the VAR model in the window
+# of data. Taken together, these represent a
+# a time-varying linear system.
+
+conn = var(data=epochs.get_data(), times=times, names=ch_names)
+
+# this returns a connectivity structure over time
+print(conn)
+
+##############################################################################
+# Evaluate this VAR model fit
+# ---------------------------
+# We can now evaluate the model fit by computing
+# the residuals of the model and visualizing it.
+# In addition, we can evaluate the covariance of the
+# residuals.
+
+predicted_data = conn.predict(epochs.get_data())
+
+# compute residuals
+residuals = epochs.get_data() - predicted_data
+
+# visualize the residuals
+fig, ax = plt.subplots()
+ax.plot(residuals.flatten(), '*')
+ax.set(
+    title='Residuals of fitted VAR model',
+    ylabel='Magnitude'
+)
+
+# compute the covariance of the residuals
+model_order = conn.attrs.get('model_order')
+t = residuals.shape[0]
+sampled_residuals = np.concatenate(
+    np.split(residuals[:, :, model_order:], t, 0),
+    axis=2
+).squeeze(0)
+rescov = np.cov(sampled_residuals)
+
+# visualize the covariance of residuals
+fig, ax = plt.subplots()
+cax = fig.add_axes([0.27, 0.8, 0.5, 0.05])
+im = ax.imshow(rescov, cmap='viridis', aspect='equal', interpolation='none')
+fig.colorbar(im, cax=cax, orientation='horizontal')
+
+##############################################################################
+# Compute one VAR model using all epochs
+# --------------------------------------
+# By setting ``model='dynamic'``, we instead treat each
+# Epoch as a sample of the same VAR model
+
+conn = var(data=epochs.get_data(), times=times, names=ch_names,
+           model='avg-epochs')
+
+# this returns a connectivity structure over time
+print(conn)
+
+##############################################################################
+# Evaluate model fit again
+# ------------------------
+# We can now evaluate the model fit again as done
+# earlier. This model fit will of course have
+# higher residuals then before as we are only
+# fitting 1 VAR model to all the epochs.
+
+first_epoch = epochs.get_data()[0, ...]
+predicted_data = conn.predict(first_epoch)
+
+# compute residuals
+residuals = epochs.get_data() - predicted_data
+
+# visualize the residuals
+fig, ax = plt.subplots()
+ax.plot(residuals.flatten(), '*')
+ax.set(
+    title='Residuals of fitted VAR model',
+    ylabel='Magnitude'
+)
+
+# compute the covariance of the residuals
+model_order = conn.attrs.get('model_order')
+t = residuals.shape[0]
+sampled_residuals = np.concatenate(
+    np.split(residuals[:, :, model_order:], t, 0),
+    axis=2
+).squeeze(0)
+rescov = np.cov(sampled_residuals)
+
+# visualize the covariance of residuals
+fig, ax = plt.subplots()
+cax = fig.add_axes([0.27, 0.8, 0.5, 0.05])
+im = ax.imshow(rescov, cmap='viridis', aspect='equal', interpolation='none')
+fig.colorbar(im, cax=cax, orientation='horizontal')
```

Comment:
Again, interpretation: what does this outcome mean?

Reply:
added.